jax/_src/interpreters/ad.py

```diff
@@ -82,6 +82,26 @@ def jvpfun(f, instantiate, transform_stack, primals, tangents):
                   in zip(out_tangents, instantiate)]
   return out_primals, out_tangents
 
+@lu.transformation_with_aux2
+def linearize_subtrace(_f, _store, _tag, nzs_in, *primals, **params):
+  with core.take_current_trace() as parent_trace:
+    tangent_trace = pe.DynamicJaxprTrace()
+    tangents = [tangent_trace.new_arg(get_aval(p).to_tangent_aval())
+                for (p, nz) in zip(primals, nzs_in) if nz]
+    linearize_trace = LinearizeTrace(parent_trace, tangent_trace, tag=_tag)
+    tracers = [LinearizeTracer(linearize_trace, p, t) for p, t in zip(primals, tangents)]
+    with core.set_current_trace(linearize_trace):
+      ans = _f(*tracers)
+  out_primals, out_tangents = unzip2(map(linearize_trace.to_primal_tangent_pair, ans))
+  nzs_out = [type(t) is not Zero for t in out_tangents]
+  out_tangents = [t for t, nz in zip(out_tangents, nzs_out) if nz]
+  out_tangents = map(tangent_trace.to_jaxpr_tracer, out_tangents)
+  jaxpr, consts, attrs_tracked = tangent_trace.to_jaxpr(out_tangents)
+  num_residuals = len(consts)
+  del attrs_tracked # TODO: attrs
```

Comment:
Raise an exception if there are any? (You can also unpack into an empty tuple to check the same, though with no explanation of what the error means.)